packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -43,6 +43,76 @@ const isInvalidFileName = (fileName: string): boolean => {
  * Returns resolved url
  */
 export async function createNewFluidFile(
+    getStorageToken: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    newFileInfo: INewFileInfo,
+    logger: ITelemetryLogger,
+    createNewSummary: ISummaryTree | undefined,
+    epochTracker: EpochTracker,
+): Promise<IOdspResolvedUrl> {
+    return createNewSummary === undefined
+        ? createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker)
+        : createNewFluidFileFromSummary(getStorageToken, newFileInfo, logger, createNewSummary, epochTracker);
+}
+
+export async function createNewEmptyFluidFile(
+    getStorageToken: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    newFileInfo: INewFileInfo,
+    logger: ITelemetryLogger,
+    epochTracker: EpochTracker,
+): Promise<IOdspResolvedUrl> {
+    // Check for valid filename before the request to create file is actually made.
+    if (isInvalidFileName(newFileInfo.filename)) {
```

Comment:
I think isInvalidFileName() check can be moved to createNewFluidFile() to reduce code duplication.
Anything else we can share between two workflows?